// ngx-plaid-link/esm2020/lib/ngx-plaid-link.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxPlaidLinkButtonComponent } from './ngx-plaid-link-button.component';
import { NgxPlaidLinkService } from './ngx-plaid-link.service';
import { NgxPlaidLinkDirective } from './ngx-plaid-link.directive';
import * as i0 from "@angular/core";
export class NgxPlaidLinkModule {
}
NgxPlaidLinkModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: NgxPlaidLinkModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxPlaidLinkModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: NgxPlaidLinkModule, declarations: [NgxPlaidLinkButtonComponent, NgxPlaidLinkDirective], imports: [CommonModule], exports: [NgxPlaidLinkButtonComponent, NgxPlaidLinkDirective] });
NgxPlaidLinkModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: NgxPlaidLinkModule, providers: [
        NgxPlaidLinkService
    ], imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: NgxPlaidLinkModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    providers: [
                        NgxPlaidLinkService
                    ],
                    declarations: [NgxPlaidLinkButtonComponent, NgxPlaidLinkDirective],
                    exports: [NgxPlaidLinkButtonComponent, NgxPlaidLinkDirective]
                }]
        }] });
